presto-hive/src/main/java/io/prestosql/plugin/hive/HiveTableHandle.java

```diff
@@ -153,6 +189,19 @@ public HiveTableHandle withAnalyzeColumnNames(Set<String> analyzeColumnNames)
                 Optional.of(analyzeColumnNames));
     }
 
+    @Override
+    public void validateScan(ConnectorSession session)
+    {
+        if (HiveSessionProperties.isQueryPartitionFilterRequired(session) && !partitionColumns.isEmpty()
+                && getEnforcedConstraint().isAll()
+                && (!getPartitionConstraint().isPresent() || !getPartitionConstraint().get().predicate().isPresent())) {
```

Comment:
could you please explain how `!getPartitionConstraint().get().predicate().isPresent()` is helping us here.
I see we are propagating constraints from `pushFilterIntoTableScan` method where the constraint is initialised like `constraint = new Constraint(newDomain);` making predicate as `Optional.empty()` since `pruneWithPredicateExpression` is `false` so shouldn't `getPartitionConstraint().get().predicate().isPresent()` will always be `false`?

Reply:
The presence of the predicate on the partition constraint that was used in applyPartitionResult is an indication that there's dynamic partition pruning occurring. The constraint passed through will have a predicate in cases in which this occurs. Did you see the test I added [here](https://github.com/prestosql/presto/pull/2334/files#r375998776)?